### PR TITLE
Update literature around changelog

### PR DESCRIPTION
**Pull request by harshavamsi**
Signed-off-by: Harsha Vamsi Kalluri <harshavamsi096@gmail.com>

### Description
Introduces changelog and changelog verifier

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [X] New functionality includes testing.
  - [X] All tests pass
- [X] New functionality has been documented.
  - [X] New functionality has comments added
  - [X] New functionality is added to `docs` folder
- [X] Commits are signed per the DCO using --signoff
- [X] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](https://github.com/opensearch-project/opensearch-java/blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/opensearch-java/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by saratvemulapalli — APPROVED** [review]

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -84,17 +84,17 @@ You may type this line on your own when writing your commit messages. However, i
...
 
-OpenSearch-Java maintains version specific changelog by enforcing a change to the ongoing [CHANGELOG](CHANGELOG.md) file adhering to the [Keep A Changelog](https://keepachangelog.com/en/1.0.0/) format.
+OpenSearch-Java maintains a version specific changelog by enforcing a change to the ongoing [CHANGELOG](CHANGELOG.md) file adhering to the [Keep a Changelog](https://keepachangelog.com/en/1.0.0/) format.
 
-Briefly, the changes are curated by version, with the changes to the main branch added chronologically to `Unreleased` version. Further, each version has corresponding sections which list out the category of the change - `Added`, `Changed`, `Deprecated`, `Removed`, `Fixed`, `Security`.
+The changes are curated by version, with the changes to the main branch added chronologically to the `Unreleased` version. Each version has corresponding sections which list out the category of the change - `Added`, `Changed`, `Deprecated`, `Removed`, `Fixed`, `Security`.
 
 
-### How to add my changes to [CHANGELOG](CHANGELOG.md)?
+### Adding changes
```
Comment (dblock) [inline CONTRIBUTING.md:92]: Should be "Adding Changes" to be consistent (capital C)

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -84,17 +84,17 @@ You may type this line on your own when writing your commit messages. However, i
 
 ## Changelog
 
-OpenSearch-Java maintains version specific changelog by enforcing a change to the ongoing [CHANGELOG](CHANGELOG.md) file adhering to the [Keep A Changelog](https://keepachangelog.com/en/1.0.0/) format.
+OpenSearch-Java maintains a version specific changelog by enforcing a change to the ongoing [CHANGELOG](CHANGELOG.md) file adhering to the [Keep a Changelog](https://keepachangelog.com/en/1.0.0/) format.
```
Comment (dblock) [inline CONTRIBUTING.md:87]: The project is Opensearch-java (capital O because start of sentence).

Version-specific I think has a `-`.

**Review by dblock — CHANGES_REQUESTED** [review]
Some more nits pls.

**Comment by harshavamsi** [comment]
@dblock updated with asks. I removed the PR template after our discussion [here](https://github.com/opensearch-project/opensearch-go/pull/172#discussion_r996064830)

**Review by VachaShah — DISMISSED** [review]

**Review by dblock — APPROVED** [review]

**Review by VachaShah — APPROVED** [review]

**Pull request merged**